### PR TITLE
Update to typing game and UI.

### PR DESCRIPTION
**Pull request by calebkrauter**
- Update start button corners and input box UI.
- Update typing font.
- Display user input as only the current word being typed and over the input box with the input text being transparent. 
- TODO - make input box text unelectable. Implement disallowing backspace after space is pressed so that only the current word can be edited and previous words may not be. Update game to end when the correct character length is reached. Implement game to end when the correct amount of words are reached rather than characters or correctness. 
- TODO - implement accuracy.
- Note: currently to end the game, you need to type the correct amount of words which currently is using the space between words to delimit the index of words such that when you reach the end of the sentence, you must press space before it will end.
- TODO- ensure that you can't press space multiple times and that you can't press space when a word has not been typed at all to prevent accuracy from being incorrect.
- TODO - implement highlighting of current word to type by only showing the current word to type.

### CODE REVIEW

File: src/views/TypingGame.vue
Comment (jasgcode) [inline src/views/TypingGame.vue:1]: [](url)

File: src/views/TypingGame.vue [hunk 1]
```diff
@@ -33,13 +45,14 @@
```
Comment (jasgcode) [inline src/views/TypingGame.vue:42]: <script>
      import { ref, computed } from 'vue';
      
      export default {
        name: 'TypingGame',
        setup() {
        
 In your iteration you have the options api style. For consistency in the codebase, update the TypingGame.vue to follow the composition api style. https://vuejs.org/guide/introduction.html#composition-api
 https://vuejs.org/guide/extras/composition-api-faq.html
 
 it should look something like this
 https://vuejs.org/guide/extras/composition-api-faq.html#smaller-production-bundle-and-less-overhead 
 
    <script setup>
      import { ref, computed } from 'vue';
        <!- Rest of code->
        <!- Remove the return statement as its not needed, refer to the link above->
            // return {
            // checkInput,
            //  handleKeydown,
            //  pasteToWin,
            //  userInputConstrained,
            //};

File: src/views/TypingGame.vue [hunk 2]
```diff
@@ -52,63 +65,77 @@
...
+        userInputConstrained.value = '';
         gameStarted.value = true;
         gameEnded.value = false;
         startTime.value = Date.now();
       }
-      let keysPressedIterator = 0;
+      const keysPressedIterator = ref(0);
       let backspacePressed = false;
 
       function checkInput() {
```
Comment (jasgcode) [inline src/views/TypingGame.vue:100]: When it comes to implementing the accuracy I would recommend you look at this documentation.
https://vuejs.org/guide/essentials/template-refs.html
https://vuejs.org/guide/essentials/reactivity-fundamentals.html
https://vuejs.org/guide/extras/reactivity-in-depth.html
https://vuejs.org/guide/scaling-up/state-management
Also perhaps a map may be better to store the typing texts for now until we have a db.

Though I would definitely recommend perhaps breaking apart this page into simpler components then using pinia to manage the state of the game.
https://pinia.vuejs.org/getting-started.html

**Review by jasgcode — APPROVED** [review]
Overall good update.

**Review by jimothy-dev — APPROVED** [review]
Looks great, good progress.

File: src/views/TypingGame.vue (hunk 1 above)
Comment (calebkrauter) [reply src/views/TypingGame.vue:42]: Thanks I'll take a look.

Comment (calebkrauter) [reply src/views/TypingGame.vue:42]: I'm working on making the change you requested. I noticed that I am having to move some of the code I wrote into a different ordering now to make it functional. For example, Ctrl+V prevention doesn't work, I need to move the code for that closer to the top of the setup and make another change or so.

Comment (calebkrauter) [reply src/views/TypingGame.vue:42]: Ok, I made the change you requested. I ran into some difficulty but it seems to be working okay now. I also update the prevention of Ctrl+V to just prevent the event from occurring which is a simpler solution to that.

**Pull request merged**